### ISSUE TITLE
Export to PNG (or PDF) can take some time for large dashboards

### ISSUE THREAD

**Issue by paoliniluis** [metabase/metabase#29750]
**Describe the bug**
If you have a large dashboard (e.g. dashboard 970 in stats), exporting a png can take as much as 5 seconds (in my 16 core ryzen 5800hs with 32 gigs of ram).

**Logs**
NA

**To Reproduce**
1) build a big big dashboard and try a png export

**Expected behavior**
This should take seconds

**Screenshots/videos**
NA

**Information about your Metabase Installation:**
- Your browser and the version: Brave latest
- Your operating system: Pop OS LTS
- Metabase version: master, but this happens in 46+

**Severity**
P3

**Additional context**
NA

**Comment by cdeweyx** [comment]
> This might hard to do, some dashboards will take a long time to serialize. Maybe we should add a mechanism similar to slow loading queries to show that it's happening in the background? cc [@cdeweyx](https://github.com/cdeweyx)

Yeah we have a pattern for that with data downloads, could definitely help with visibility here. Ideally could also help with https://github.com/metabase/metabase/issues/33263 which is different, but adjacent to this.

**Comment by perivamsi** [comment]
@paoliniluis do you still see this? also, how do you export a dashboard as png? I can only see pdf option.

**Comment by dragonsahead** [comment]
yup, try dashboard 6424

**Comment by iethree** [comment]
speeding this up is a Big Job. we can improve the UX significantly by showing a loading indicator

**Comment by ixipixi** [comment]
@eliotmetabase @iethree download performance seems to be significantly worse on v58 than v57.

**Comment by perivamsi** [comment]
Upgrading this to p1 so we can investigate the performance regression between 57 and 58

**Comment by dragonsahead** [comment]
Adding a performance trace from a customer https://drive.google.com/file/d/1iLZzX1kj4UMZVIWJUdfN1xIFN9SWaWuS/view?usp=drive_link

**Comment by dragonsahead** [comment]
closing in favor of https://github.com/metabase/metabase/issues/60460 as it has more info
